test/unit/sql/delegatesList.js

```diff
@@ -0,0 +1,451 @@
+				done('Should not pass');
+			}).catch(function (err) {
+				expect(err).to.be.an('error');
+				expect(err.message).to.contain('Invalid parameters supplied');
+				done();
+			});
+		});
+
+		it('SQL generateDelegatesList() should raise exception for negative round', function (done) {
+			var round = -1;
+			var delegates = ['1'];
+
+			db.query(sql.generateDelegatesList, {round: round, delegates: delegates}).then(function (rows) {
+				done('Should not pass');
+			}).catch(function (err) {
+				expect(err).to.be.an('error');
+				expect(err.message).to.contain('Invalid parameters supplied');
+				done();
+			});
+		});
+
+		it('SQL generateDelegatesList() should raise exception for empty delegates', function (done) {
+			var round = 1;
+			var delegates = [];
+
+			db.query(sql.generateDelegatesList, {round: round, delegates: delegates}).then(function (rows) {
+				done('Should not pass');
+			}).catch(function (err) {
+				expect(err).to.be.an('error');
+				expect(err.message).to.contain('cannot determine type of empty array');
+				done();
+			});
+		});
+
+		it('SQL generateDelegatesListCast() should raise exception for empty delegates', function (done) {
+			var round = 1;
+			var delegates = [];
+
+			db.query(sql.generateDelegatesListCast, {round: round, delegates: delegates}).then(function (rows) {
+				done('Should not pass');
+			}).catch(function (err) {
+				expect(err).to.be.an('error');
+				expect(err.message).to.contain('Invalid parameters supplied');
+				done();
+			});
+		});
+
+		it('SQL generateDelegatesList() should raise exception for null delegates', function (done) {
+			var round = 1;
+			var delegates = null;
+
+			db.query(sql.generateDelegatesList, {round: round, delegates: delegates}).then(function (rows) {
+				done('Should not pass');
+			}).catch(function (err) {
+				expect(err).to.be.an('error');
+				expect(err.message).to.contain('Invalid parameters supplied');
+				done();
+			});
+		});
+
+		it('SQL generateDelegatesList() should raise exception for null delegates', function (done) {
+			var round = 1;
+			var delegates = undefined;
+
+			db.query(sql.generateDelegatesList, {round: round, delegates: delegates}).then(function (rows) {
+				done('Should not pass');
+			}).catch(function (err) {
+				expect(err).to.be.an('error');
+				expect(err.message).to.contain('Invalid parameters supplied');
+				done();
+			});
+		});
+	});
+
+	describe('checking SQL function getDelegatesList()', function () {
+		it('SQL getDelegatesList() results should be equal to generateDelegatesList() - real 101 delegates from current database', function (done) {
+			db.task(function (t) {
+				return t.batch([
+					t.query(sql.getDelegatesList),
+					t.query(sql.getActiveDelegates),
+					t.query(sql.getRound)
+				]);
+			}).then(function (res) {
+				var delegates_list   = res[0][0].list,
+					active_delegates = res[1][0].delegates,
+					round            = res[2][0].round;
```

Comment:
There is nothing wrong with formatting here - it's how github displays it. ESLint require here 5 tabs, and it's good practice to align to equal sign when dealing with multiple assignments. However I'll change to separate `var` each line, as @karmacoma suggested,